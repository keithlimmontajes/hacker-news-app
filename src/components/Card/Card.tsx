import React from 'react';
import {View} from 'tamagui';
import {CardProps} from './types';

const Card = ({children}: CardProps) => {
  return (
    <View
      my="$2"
      bg="white"
      p={'$2'}
      borderWidth={1}
      borderRadius={8}
      // @ts-ignore
      borderColor="$borderGray">
      {children}
    </View>
  );
};

export default Card;
