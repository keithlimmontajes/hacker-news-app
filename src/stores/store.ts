import {configureStore} from '@reduxjs/toolkit';
import {persistStore, persistReducer} from 'redux-persist';
import {apiMiddleware} from './middleware';
import rootReducer from './rootReducer';
import AsyncStorage from '@react-native-async-storage/async-storage';

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  whitelist: ['newsSlice'],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: ['persist/PERSIST', 'persist/REHYDRATE'],
      },
    }).concat(apiMiddleware),
});

export const persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
