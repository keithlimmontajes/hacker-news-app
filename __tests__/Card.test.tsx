import React from 'react';
import {render} from '@testing-library/react-native';
import Card from '../src/components/Card';
import {Provider} from 'react-redux';
import store from '../src/stores/store';
import {NavigationContainer} from '@react-navigation/native';
import {TamaguiProvider} from 'tamagui';
import config from '../src/themes';
import {ThemeProvider, View} from '@tamagui/core';

describe('CardComponent', () => {
  it('Should work as expected', () => {
    const all = render(
      <Provider store={store}>
        <ThemeProvider defaultTheme="light">
          <NavigationContainer>
            <TamaguiProvider config={config}>
              <Card>
                <View />
              </Card>
            </TamaguiProvider>
          </NavigationContainer>
        </ThemeProvider>
      </Provider>,
    );
    expect(all.toJSON()).toMatchSnapshot();
  });
});
