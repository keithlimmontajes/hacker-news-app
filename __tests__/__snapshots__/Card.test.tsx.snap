// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent Should work as expected 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "borderBottomColor": "#E9E9E9",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderBottomWidth": 1,
      "borderLeftColor": "#E9E9E9",
      "borderLeftWidth": 1,
      "borderRightColor": "#E9E9E9",
      "borderRightWidth": 1,
      "borderStyle": "solid",
      "borderTopColor": "#E9E9E9",
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "borderTopWidth": 1,
      "marginBottom": 7,
      "marginTop": 7,
      "paddingBottom": 7,
      "paddingLeft": 7,
      "paddingRight": 7,
      "paddingTop": 7,
    }
  }
>
  <View />
</View>
`;
